flocker/control/httpapi.py

```diff
@@ -185,6 +185,143 @@ def saved(ignored):
         saving.addCallback(saved)
         return saving
 
+    @app.route("/datasets/<dataset_id>", methods=['POST'])
+    @user_documentation(
+        """
+        Update an existing dataset.
+        """,
+        examples=[
+            u"update dataset with primary",
+        ]
+    )
+    @structured(
+        # XXX Validation of the supplied primary address is performed by input
+        # schema validation. Any further checks required?
+        inputSchema={'$ref': '/v1/endpoints.json#/definitions/datasets'},
+        outputSchema={'$ref': '/v1/endpoints.json#/definitions/datasets'},
+        schema_store=SCHEMAS
+    )
+    def update_dataset(self, dataset_id, primary=None, maximum_size=None,
+                       metadata=None):
+        """
+        Update an existing dataset in the cluster configuration.
+
+        :param unicode dataset_id: The unique identifier of the dataset.  This
+            is a string giving a UUID (per RFC 4122).
+
+        :param unicode primary: The address of the node to which the dataset
+            will be moved.
+
+        :param int maximum_size: The maximum number of bytes the dataset will
+            be capable of storing.  This may be optional or required depending
+            on the dataset backend.
+
+        :param dict metadata: A small collection of unicode key/value pairs to
+            associate with the dataset.  These items are not interpreted.  They
+            are only stored and made available for later retrieval.  Use this
+            for things like human-friendly dataset naming, ownership
+            information, etc.
+
+        :return: A ``dict`` describing the dataset which has been added to the
+            cluster configuration or giving error information if this is not
+            possible.
+        """
+        # XXX Changing volume size and metadata via the API is beyond the scope
+        # of this issue. Decide how to react.
```

Comment:
`if maximum_size is None` is the wrong test... since semantically `None` means "no maximum".
The solution is to change the default argument in `update_dataset()` to be some placeholder object (`NO_ARGUMENT = object()` or something) so you can distinguish between missing argument and a value which is `None`.